grails-app/services/org/broadinstitute/orsp/CommentsService.groovy

```diff
@@ -0,0 +1,84 @@
+     */
+    Comment addComment(String issueId, String commentToSave) throws IllegalArgumentException{
+        if (commentToSave) {
+            Issue issue = queryService.findByKey(issueId)
+            Comment comment = persistenceService.saveComment(issue.projectKey, getUser()?.displayName, commentToSave)
+
+            if (comment == null) {
+                log.error("Error saving comment for issue '" + issueId + "': null")
+                throw new IllegalArgumentException("Error trying to save comment.")
+            }
+            if (comment.hasErrors()) {
+                comment.errors.getAllErrors().each {
+                    log.error("Error saving comment for issue '" + issueId + "': " + it)
+                }
+                throw new Exception("Error trying to save comment.")
+            }
+
+            // By default, comments should go to ORSP
+            List<String> toAddresses = new ArrayList<>()
+            toAddresses.addAll(notifyService.getOrspSpecialRecipients())
+            String fromAddress = (String) getUser()?.emailAddress
+
+            // If the user is ORSP/Admin, and there are PMs available, send the comment to the PMs.
+            if (isAdmin() && !issue.getPMs().isEmpty()) {
+                toAddresses?.clear()
+                toAddresses.addAll(userService.findUsers(issue.getPMs())?.collect {it.emailAddress})
+                fromAddress = notifyService.ORSP_ADDRESS
+            }
+
+            // Extra check to cover error condition of empty PMs
+            if (toAddresses.isEmpty()) {
+                log.error("Issue " + issue.projectKey + " has empty PMs.")
+                toAddresses.addAll(notifyService.getOrspSpecialRecipients())
+            }
+
+            notifyService.sendComment(
+                    new NotifyArguments(
+                            toAddresses: toAddresses,
+                            fromAddress: fromAddress,
+                            subject: "Comment Entered: " + issue.projectKey,
+                            comment: comment.description,
+                            user: getUser(),
+                            issue: issue))
+            comment
+        } else {
+            throw new IllegalArgumentException("Empty comment to save")
+        }
+    }
+
+    /**
+     * @param issueId Project or Data Sample Cohort id
+     * @return  Collection of Comments associated to the given issueId
+     */
+    Collection<Comment> getCommentsForIssueId(String issueId) {
+        if (issueId) {
+            try {
+                Collection<Comment> comments = queryService.getCommentsByIssueId(issueId)
+                comments
+            } catch (Exception e) {
+                log.error("An error has occurred when trying to get comments for issueId: ${issueId}.", e)
+                throw new Error()
+            }
+        } else {
+            log.error("Unable to get Comments from null issueId.")
+            throw new IllegalArgumentException()
```

Comment:
nit: Should have something in the exception, something like `Issue id is required.`